subsys/bluetooth/services/ots/ots_client.c

```diff
@@ -1185,6 +1291,53 @@ int bt_ots_client_read_object_data(struct bt_ots_client *otc_inst,
 	return oacp_read(conn, inst);
 }
 
+int bt_ots_client_write_object_data(struct bt_ots_client *otc_inst,
+				    struct bt_conn *conn, const void *buf, size_t len,
+				    off_t offset, enum bt_ots_oacp_write_mode mode)
+{
+	struct bt_otc_internal_instance_t *inst;
+
+	CHECKIF(!conn) {
+		BT_WARN("Invalid Connection");
+		return -ENOTCONN;
+	}
+	CHECKIF(!otc_inst) {
+		BT_ERR("Invalid OTC instance");
+		return -EINVAL;
+	}
+	CHECKIF((mode != BT_OTS_OACP_WRITE_OP_MODE_TRUNCATE) &&
+		(mode != BT_OTS_OACP_WRITE_OP_MODE_NO_TRUNCATE)) {
+		BT_ERR("Invalid write object mode parameter %d", mode);
+		return -EINVAL;
+	}
+
+	/* OTS_v10.pdf Table 3.9: Object Action Control Point Procedure Requirements
+	 *	Offset and Length field are UINT32 Length
+	 */
+	CHECKIF(len > UINT32_MAX) {
+		BT_ERR("length exceed UINT32");
+		return -EINVAL;
+	}
+	CHECKIF(offset > UINT32_MAX) {
+		BT_ERR("offset exceed UINT32");
+		return -EINVAL;
+	}
+
+	inst = lookup_inst_by_handle(otc_inst->start_handle);
+
+	if (!inst) {
+		BT_ERR("Invalid OTC instance");
+		return -EINVAL;
+	}
+
+	if ((len > (otc_inst->cur_object.size.alloc - otc_inst->cur_object.size.cur)) ||
+		len == 0) {
+		BT_ERR("Invalid write len: %zu", len);
+		return -EINVAL;
+	}
+	return oacp_write(conn, inst, buf, len, offset, mode);
```

Comment:
```suggestion
	return oacp_write(conn, inst, buf, (uint32_t)len, (uint32_t)offset, mode);
```
And then update `oacp_write` to use `uint32_t` instead of `size_t` and `off_t`.

Reply:
Change internal function data type to uint32_t.